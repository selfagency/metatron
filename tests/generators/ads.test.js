const generateAds = require('../../dist/generators/ads').default

test('generate ads.txt - static', () => {
  const adsTxt = generateAds({
    settings: {
      mode: 'generator',
      output_dir: './'
    },
    ads: {
      sellers: [
        {
          domain: 'example.com',
          publisher_id: 'abc1234',
          account_type: 'DIRECT'
        },
        {
          domain: 'example.com',
          publisher_id: 'abc1234',
          account_type: 'DIRECT',
          cert_auth_id: '1234abc'
        },
        {
          domain: 'example.com',
          publisher_id: 'abc1234',
          account_type: 'DIRECT',
          comment: 'this is a comment'
        }
      ],
      inventory_partners: ['example.com', 'test.com'],
      subdomains: ['ads.example.com', 'ads.test.com'],
      contact: 'adops@example.com'
    }
  })

  expect(adsTxt).toMatchSnapshot()
})

// test('generate ads.txt - middleware', () => {})
