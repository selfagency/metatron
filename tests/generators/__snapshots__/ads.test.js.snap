// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate ads.txt - static 1`] = `
"example.com, abc1234, DIRECT
example.com, abc1234, DIRECT, 1234abc
example.com, abc1234, DIRECT # this is a comment
inventorypartnerdomain=example.com
inventorypartnerdomain=test.com
subdomain=ads.example.com
subdomain=ads.test.com
contact=adops@example.com

# Made with Metatron - https://github.com/selfagency/metatron
"
`;
